src/python-common/ceph/deployment/service_spec.py

```diff
@@ -817,6 +817,7 @@ def __init__(self,
                  unmanaged: bool = False,
                  preview_only: bool = False,
                  networks: Optional[List[str]] = None,
+                 targets: Optional[List[str]] = None,
```

Comment:
i see that we are adding the targets in the service_spec but how's the target getting set in the config when the prometheus is deployed? I don't see that here..  All I see is a command that sets the target which is not run when the service is created!

Reply:
@nizamial09 , the scope of this PR is to add the targets via cli commands..Adding the targets on prometheus service deployment can be handled in a separate PR

Reply:
@aaSharma14 if the scope is just to add the targets via cli commands, then do we need this in the service_spec now in this PR?

Reply:
It would be out of the norm for us to merge a new spec field without it having any functionality FWIW. But it looks like without it, a whole bunch of the rest of this PR wouldn't work either (multiple places looking at the `targets` attribute of the prometheus spec). What else do we need beyond just adjusting the scrape config template and setting up the targets there for prometheus to make use of this? If it's not a massive change, maybe it should be included in this PR as well.

Reply:
Hey @adk3798 @nizamial09 , I tried creating a prometheus service with a spec by including the targets field in the spec file and I was able to create the service with the desired targets in place in the prometheus configuration. I think this is the intended behavior or Do I need to do anything else here in this PR? Thanks

Reply:
> Hey @adk3798 @nizamial09 , I tried creating a prometheus service with a spec by including the targets field in the spec file and I was able to create the service with the desired targets in place in the prometheus configuration. I think this is the intended behavior or Do I need to do anything else here in this PR? Thanks

I don't know much about the feature so hard for me to say. Was getting the targets list into the prometheus conf the end goal?

Reply:
Yes, getting the targets list into the prometheus conf and adding an external label `cluster` which contains the cluster fsid to the prometheus metrics is the desired goal from this PR.

Reply:
To elaborate, this task is necessary for multi-cluster monitoring. We take the Prometheus endpoints of the external clusters as input from the user and add these targets to the Prometheus configuration of the hub cluster. This allows us to obtain the combined metrics of all the external clusters that are connected. This PR is meant to expose CLI commands for adding/removing a target from the Prometheus configuration.